source/plugins/go/src/oms.go

```diff
@@ -1177,51 +1179,82 @@ func PostInputPluginRecords(inputPluginRecords []map[interface{}]interface{}) in
+				ContainerLogTelemetryMutex.Lock()
+				defer ContainerLogTelemetryMutex.Unlock()
+				InputPluginRecordsErrors += 1
+			}
+
+			var bts int
+			var er error
+			var elapsed time.Duration
+			deadline := 10 * time.Second
+
+			if !IsWindows {
+				if MdsdInputPluginRecordsMsgpUnixSocketClient != nil {
+					MdsdInputPluginRecordsMsgpUnixSocketClient.SetWriteDeadline(time.Now().Add(deadline)) //this is based of clock time, so cannot reuse
+					bts, er = MdsdInputPluginRecordsMsgpUnixSocketClient.Write(msgpBytes)
+					elapsed = time.Since(start)
+				} else {
 					Log("Error::mdsd::Unable to create mdsd client for InputPluginRecords. Please check error log.")
-					ContainerLogTelemetryMutex.Lock()
-					defer ContainerLogTelemetryMutex.Unlock()
-					InputPluginRecordsErrors += 1
 				}
 			}
-			if MdsdInputPluginRecordsMsgpUnixSocketClient != nil {
-				deadline := 10 * time.Second
-				MdsdInputPluginRecordsMsgpUnixSocketClient.SetWriteDeadline(time.Now().Add(deadline)) //this is based of clock time, so cannot reuse
-				bts, er := MdsdInputPluginRecordsMsgpUnixSocketClient.Write(msgpBytes)
-				elapsed := time.Since(start)
-				if er != nil {
-					message := fmt.Sprintf("Error::mdsd::Failed to write to input plugin mdsd %d records after %s. Will retry ... error : %s", len(msgPackEntries), elapsed, er.Error())
-					Log(message)
-					if MdsdInputPluginRecordsMsgpUnixSocketClient != nil {
-						MdsdInputPluginRecordsMsgpUnixSocketClient.Close()
-						MdsdInputPluginRecordsMsgpUnixSocketClient = nil
-					}
-					SendException(message)
+
+			if IsWindows {
+				if InputPluginNamedPipe != nil {
+					InputPluginNamedPipe.SetWriteDeadline(time.Now().Add(deadline)) //this is based of clock time, so cannot reuse
+					bts, er = InputPluginNamedPipe.Write(msgpBytes)
+					elapsed = time.Since(start)
 				} else {
-					telemetryDimensions := make(map[string]string)
-					lowerTag := strings.ToLower(tag)
-					if strings.Contains(lowerTag, strings.ToLower(ContainerInventoryDataType)) {
-						telemetryDimensions["ContainerInventoryCount"] = strconv.Itoa(len(msgPackEntries))
-					} else if strings.Contains(lowerTag, strings.ToLower(InsightsMetricsDataType)) {
-						telemetryDimensions["InsightsMetricsCount"] = strconv.Itoa(len(msgPackEntries))
-					} else if strings.Contains(lowerTag, strings.ToLower(PerfDataType)) {
-						telemetryDimensions["PerfCount"] = strconv.Itoa(len(msgPackEntries))
-					} else {
-						Log("FlushInputPluginRecords::Warn::Flushed records of unknown data type %s", lowerTag)
-					}
-					numRecords := len(msgPackEntries)
-					Log("FlushInputPluginRecords::Info::Successfully flushed %d records that was %d bytes in %s", numRecords, bts, elapsed)
-					// Send telemetry to AppInsights resource
-					SendEvent(InputPluginRecordsFlushedEvent, telemetryDimensions)
+					Log("Error::mdsd::Unable to create mdsd client for InputPluginRecords. Please check error log.")
 				}
```

Comment:
can we refactor this? looks like we have multiple instances of if Linux do this and if not do that. Instead is it possible to have one block of if Linux and one block for if Windows?

Reply:
That would mean a lot of code which is similar in both have to be repeated multiple times in the code? Are we okay with that ?

Reply:
good pt. I'll leave it upto you. If its possible to refactor it to make it more readible please do it otherwise its ok.

Reply:
Refactored the code